templates/common/_base/units/nodeip-configuration.service.yaml

```diff
@@ -0,0 +1,44 @@
+name: nodeip-configuration.service
+enabled: {{if eq .Infra.Status.PlatformStatus.Type "None"}}true{{else}}false{{end}}
+contents: |
+  [Unit]
+  Description=Writes IP address configuration so that kubelet and crio services select a valid node IP
+  Wants=network-online.target
+  After=network-online.target ignition-firstboot-complete.service
+  Before=kubelet.service crio.service
+
+  [Service]
+  # Need oneshot to delay kubelet
+  Type=oneshot
+  # Would prefer to do Restart=on-failure instead of this bash retry loop, but
+  # the version of systemd we have right now doesn't support it. It should be
+  # available in systemd v244 and higher.
+  ExecStart=/bin/bash -c " \
```

Comment:
We also need to de-dedup this bit but in other code we've explicitly used a retry loop for the pull phase, but not for the run phase to more cleanly distinguish between the two.

That said perhaps in this case we do want to retry always?

Reply:
Not sure if you're saying I should change something here or not? FWIW this is just copied from the other `nodeip-configuration.service.yaml`'s and is currently identical to what they're doing.

Also, note that we're passing `--retry-on-failure` to `runtimecfg node-ip` here, so it's _also_ looping (forever, unless it gets an error from netlink, which is pretty "can't happen" and likely won't be fixed by a retry at any level)